test/passing/module.ml

```diff
@@ -28,3 +28,41 @@ module C = struct
 
   and B : sig end = B
 end
+
+let _ =
+  ( module M : S
+    with type long_long_type = t
+     and type long_long_long_type = u
+     and type very_long_long_type = t )
+
+let _ =
+  ( module struct
+      type t
+    end : S with type t = t )
```

Comment:
Here a phrase is placed at the end of the last line of the preceding phrase which spans multiple lines. This is something that the rest of the formatting tries hard not to do. Maybe here an exception is ok, since `end` will not ever have anything before it on the same line. But can you just confirm that this is intended?

Reply:
This is intended but not required.

Reply:
(Before `:`, not `with`.) I think that for these where the module type expression fits on a line, it could be placed after the `end` as here. But once the module type expression does not fit on a line, we need a break before `:`. So I think that the question is whether it is worth having what seems to be two different layouts. I wouldn't object to having two layouts, but in the past others have been more hesitant.

Reply:
I'll implement a different layout.